pkg/webui/console/containers/access-point-list/index.js

```diff
@@ -0,0 +1,230 @@
+    'This list shows WiFi networks as detected by your gateway. You can select an access point or choose "Other..." to enter an SSID of a hidden access point.',
+  scanningWifi: 'Scanning WiFi networks…',
+})
+
+const computeDeltaInSeconds = (from, to) => {
+  // Avoid situations when server clock is ahead of the browser clock.
+  if (from > to) {
+    return 0
+  }
+
+  return Math.floor((from - to) / 1000)
+}
+
+const wifiIconBasedOnRSSI = rssi => {
+  if (rssi >= -60) {
+    // Strong
+    return 'wifi'
+  } else if (rssi >= -80) {
+    // Moderate
+    return 'signal_wifi_2'
+  }
+  // Weak
+  return 'signal_wifi_1'
+}
+
+const AccessPointListItem = ({ accessPoint, onClick, isActive }) => {
+  const handleClick = useCallback(() => {
+    onClick(accessPoint)
+  }, [accessPoint, onClick])
+  const isOther = accessPoint.type === 'other'
+
+  return (
+    <div
+      className={classnames(style.item, 'd-flex al-center j-between', {
+        [style.active]: isActive,
+      })}
+      onClick={handleClick}
+    >
+      <div className="d-flex al-center gap-cs-xs">
+        {!isOther && <Icon icon={wifiIconBasedOnRSSI(accessPoint.rssi)} />}
+        {isOther ? <Message content={sharedMessages.otherOption} /> : accessPoint.ssid}
+      </div>
+      {accessPoint.authentication_mode !== 'open' && !isOther && (
+        <Icon icon="lock" small className="tc-subtle-gray" />
+      )}
+    </div>
+  )
+}
+
+AccessPointListItem.propTypes = {
+  accessPoint: PropTypes.accessPoint.isRequired,
+  isActive: PropTypes.bool.isRequired,
+  onClick: PropTypes.func.isRequired,
+}
+
+const AccessPointList = ({ onChange, value, className, inputWidth, onBlur, ssid }) => {
+  const [lastRefresh, setLastRefresh] = useState(undefined)
+
+  const dispatch = useDispatch()
+  const isLoading = useSelector(state => selectFetchingEntry(state, GET_ACCESS_POINTS_BASE))
+  const accessPoints = useSelector(selectAccessPoints)
+  const selectedGateway = useSelector(selectSelectedGateway)
+  const { ids } = selectedGateway
+  const isFirstRender = useRef(true)
+
+  const [isMounted, setIsMounted] = useState(true)
+
+  const handleScanAccessPoints = useCallback(() => {
+    dispatch(attachPromise(getAccessPoints(ids.gateway_id, ids.eui))).then(() => {
+      if (isMounted) {
+        setLastRefresh(new Date())
+      }
+    })
+  }, [dispatch, ids.eui, ids.gateway_id, isMounted])
+
+  // Trigger this useEffect only the first time component is rendered and has accessPoints loaded.
+  // Change value only if ssid is provided. (if it's edit form)
+  useEffect(() => {
+    if (!isFirstRender.current && ssid && !isLoading) {
+      const accessPoint = accessPoints.find(ap => ap.ssid === ssid)
+      const updatedAccessPoint = {
+        ...accessPoint,
+        is_password_set: true,
+        type: accessPoint ? 'all' : 'other',
+      }
+      onChange(updatedAccessPoint, true)
+    }
+    isFirstRender.current = false
+    /* eslint-disable */
+  }, [accessPoints, isLoading])
```

Comment:
```suggestion
  const hasSsid = Boolean(ssid)
  // Trigger this useEffect only the first time component is rendered and has accessPoints loaded.
  // Change value only if ssid is provided. (if it's edit form)
  useEffect(() => {
    if (!isFirstRender.current && hasSsid && !isLoading) {
      const accessPoint = accessPoints.find(ap => ap.ssid === ssid)
      const updatedAccessPoint = {
        ...accessPoint,
        is_password_set: true,
        type: accessPoint ? 'all' : 'other',
      }
      onChange(updatedAccessPoint, true)
    }
    isFirstRender.current = false
  }, [accessPoints, isLoading, hasSsid, onChange])
```

Haven't tested but wouldn't this do the same without disabling the dependency array?

Reply:
The `onChange` reference seems to be recreated and enters a loop which freezes the UI.

Reply:
But that's an issue in itself. Seems like there's maybe some array or object without referential equality in its dependency array maybe.

Reply:
`encodedValue` dependency in `FormField` is causing the infinite loop. Probably this function needs to be refactored smh.

```
const handleChange = useCallback(
    async (value, enforceValidation = false) => {
      const oldValue = encodedValue
      const newValue = encode(extractValue(value), oldValue)
      let isSyntheticEvent = false

      if (isPlainObject(value)) {
        // Check if the value is react's synthetic event.
        isSyntheticEvent = 'target' in value

        // TODO: Remove `await` and event persist when https://github.com/jaredpalmer/formik/issues/2457
        // is resolved.
        if (typeof value.persist === 'function') {
          value.persist()
        }
      }

      // This middleware takes care of updating the form values and allows for more control
      // over how the form values are changed if needed. See the default prop to understand
      // how the value is set by default.
      await valueSetter(
        { setFieldValue, setValues, setFieldTouched },
        { name, names, value: newValue, oldValue },
      )

      if (enforceValidation) {
        for (const name of names) {
          setFieldTouched(name, true, true)
        }
      }

      onChange(isSyntheticEvent ? value : encode(value, oldValue))
    },
    [
      encode,
      encodedValue,
      name,
      names,
      onChange,
      setFieldTouched,
      setFieldValue,
      setValues,
      valueSetter,
    ],
  )
```